### PR TITLE
fix(dg/koji): add flowchart for precheck

### LINKED ISSUE #826: Clarify allowed_commiters
In the documentation of `allowed_commiters`, it is unclear who is a committer. 

https://packit.dev/docs/configuration/downstream/koji_build#optional-parameters

Is it the author? Committer? How do we differentiate it?

What is especially not clear to the user is what will happen when the commit from `packit` user is manually merged to  the different branch. Will it trigger a build?

Either more explicit documentation or (and) corner case example with comments will be helpful.

### PR DESCRIPTION
**Pull request by mfocko**
Add flowchart that should be explicit about the way Packit decides whether to run or skip Koji build when a dist-git event is registered.

Fixes #826

### CODE REVIEW

File: docs/configuration/downstream/koji_build.md [hunk 1]
```diff
@@ -76,6 +76,32 @@ Packit dist-git configuration.
...
+```mermaid
+flowchart TD;
+  A[Push event is registered] --> B{"Is build configured
+  for the branch?"}
+  B -. No  .-> NOK((Skip build))
+  B -- Yes --> C{"Has commit been
+  pushed by Pagure?"}
+  C -- Yes --> E{"Is it a PR merge?"}
+  C -. No  .-> D{"Has the commit been pushed
+  by someone in <code>allowed_committers</code>?"}
```
Comment (mfocko) [inline docs/configuration/downstream/koji_build.md:91]: These got rendered at least locally for me, so it shouldn't be a problem, just GitHub preview is screwed.

File: docs/configuration/downstream/koji_build.md [hunk 2]
```diff
@@ -76,6 +76,32 @@ Packit dist-git configuration.
...
+  D -. No  .-> NOK
+  D -- Yes --> OK(((Run build)))
+  E -. No  .-> NOK
+  E -- Yes --> F{Specfile changes}
+  F -. No  .-> NOK
+  F -- Yes --> G{"Is the PR author in
+  <code>allowed_pr_authors</code>?"}
```
Comment (xsuchy) [inline docs/configuration/downstream/koji_build.md:98]: Is it actually true? I think we now check who is the author of the last commit in the PR.

Comment (lbarcziova) [reply docs/configuration/downstream/koji_build.md:98]: no, this is correct

**Review by lbarcziova — COMMENTED** [review]
thanks, it is much clearer from this!

But the specfile change is checked in both cases (just that currently, for pushes this pre-filtering is already done in fedmsg listener)

**Comment by mfocko** [comment]
> thanks, it is much clearer from this!
> 
> But the specfile change is checked in both cases (just that currently, for pushes this pre-filtering is already done in fedmsg listener)

I think I can simplify it with that… also adding the comment trigger

**Review by lbarcziova — APPROVED** [review]
thanks!

(I guess I will need to update this in #830 once this PR is merged :D)

**Pull request merged**